internal/mode/static/state/graph/policies_test.go

```diff
@@ -508,6 +539,104 @@ func TestAttachPolicyToGateway(t *testing.T) {
 	}
 }
 
+func TestAttachPolicyToService(t *testing.T) {
+	t.Parallel()
+
+	winningGwNsName := types.NamespacedName{Namespace: testNs, Name: "gateway"}
+	ignoredGwNsName := types.NamespacedName{Namespace: testNs, Name: "ignored-gateway"}
+
+	getGateway := func(valid bool) *Gateway {
+		return &Gateway{
+			Source: &v1.Gateway{
+				ObjectMeta: metav1.ObjectMeta{
+					Name:      winningGwNsName.Name,
+					Namespace: winningGwNsName.Namespace,
+				},
+			},
+			Valid: valid,
+		}
+	}
+
+	tests := []struct {
+		policy       *Policy
+		svc          *ReferencedService
+		gw           *Gateway
+		name         string
+		expAncestors []PolicyAncestor
+		expAttached  bool
+	}{
+		{
+			name:        "attachment",
+			policy:      &Policy{Source: &policiesfakes.FakePolicy{}},
+			svc:         &ReferencedService{ParentGateways: []types.NamespacedName{winningGwNsName}},
+			gw:          getGateway(true /*valid*/),
+			expAttached: true,
+			expAncestors: []PolicyAncestor{
+				{
+					Ancestor: getGatewayParentRef(winningGwNsName),
+				},
+			},
+		},
+		{
+			name:        "attachment; multiple parent refs - one is winning gateway",
+			policy:      &Policy{Source: &policiesfakes.FakePolicy{}},
+			svc:         &ReferencedService{ParentGateways: []types.NamespacedName{ignoredGwNsName, winningGwNsName}},
+			gw:          getGateway(true /*valid*/),
+			expAttached: true,
+			expAncestors: []PolicyAncestor{
+				{
+					Ancestor: getGatewayParentRef(winningGwNsName),
+				},
+			},
+		},
+		{
+			name:         "no attachment; parent gateway is not winning gateway",
+			policy:       &Policy{Source: &policiesfakes.FakePolicy{}},
+			svc:          &ReferencedService{ParentGateways: []types.NamespacedName{ignoredGwNsName}},
+			gw:           getGateway(true /*valid*/),
+			expAttached:  false,
+			expAncestors: nil,
+		},
+		{
+			name:        "no attachment; gateway is invalid",
+			policy:      &Policy{Source: &policiesfakes.FakePolicy{}},
+			svc:         &ReferencedService{ParentGateways: []types.NamespacedName{winningGwNsName}},
+			gw:          getGateway(false /*valid*/),
+			expAttached: false,
+			expAncestors: []PolicyAncestor{
+				{
+					Ancestor:   getGatewayParentRef(winningGwNsName),
+					Conditions: []conditions.Condition{staticConds.NewPolicyTargetNotFound("Parent Gateway is invalid")},
+				},
+			},
+		},
+		{
+			name:         "no attachment; max ancestor",
+			policy:       &Policy{Source: createTestPolicyWithAncestors(16)},
+			svc:          &ReferencedService{ParentGateways: []types.NamespacedName{winningGwNsName}},
+			gw:           getGateway(true /*valid*/),
+			expAttached:  false,
+			expAncestors: nil,
+		},
+	}
+
+	for _, test := range tests {
+		t.Run(test.name, func(t *testing.T) {
+			t.Parallel()
+			g := NewWithT(t)
+
+			attachPolicyToService(test.policy, test.svc, test.gw, "ctrl")
```

Comment:
```suggestion
			attachPolicyToService(test.policy, test.svc, test.gw, "ctlr")
```
Age old pet peeve of mine haha

Reply:
lol I ALWAYS get this wrong